### PR TITLE
Added menu bar in MainForm.

### PR DESCRIPTION
**Pull request by mllrdev-carlo-medel**
This commit will also make the adding
of items to be independent compared to
the previous implementation. Also removed
the generation of primary key from the
application layer, it is now handled by the
database.

### CODE REVIEW

File: Forms/AddItemsForm.cs [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                return itemPromptForm.IsSuccess ? itemPromptForm.Item : null;
+            }
+            catch (Exception ex)
+            {
+                Logger.log.Error(ex.ToString());
+                return null;
+            }
+        }
+
+        private void SearchButton_Click(object sender, EventArgs e)
```
Comment (KarlZablan) [inline Forms/AddItemsForm.cs:135]: Rework. Why should I fetch all records to search?

File: Forms/CartForm.cs [hunk 2]
```diff
@@ -195,8 +195,8 @@ private void RemoveButton_Click(object sender, EventArgs e)
                                     itemsForm.LoadData();
                                 }
```
Comment (KarlZablan) [inline Forms/CartForm.cs]: Rework

File: Forms/ItemsForm.cs [hunk 3]
```diff
@@ -42,67 +42,24 @@ public void LoadData()
 
                 foreach (Item item in _itemManager.GetAll())
                 {
-                    ListViewItem listViewItem = new ListViewItem(new[] {string.Empty,
+                    if (item.Stocks > 0)
```
Comment (KarlZablan) [inline Forms/ItemsForm.cs:45]: Same comment with AddItemForm's Search functionality

File: Forms/ProfileForm.cs [hunk 4]
```diff
@@ -58,11 +58,21 @@ public void LoadData()
 
                 foreach (PurchaseHistory purchaseHistory in _customer.PurchaseHistory.OrderByDescending(x => DateTime.Parse(x.Purchase.Date)))
                 {
+                    if (purchaseHistory.Purchase.Status == ProfileStringConstants.PENDING.FirstCharToUpper())
```
Comment (KarlZablan) [inline Forms/ProfileForm.cs]: don't manipulate constants

**Review by KarlZablan — CHANGES_REQUESTED** [review]
Please Check changes

File: Forms/AddItemsForm.cs [hunk 5]
```diff
@@ -189,5 +189,46 @@ private void AddItemsForm_FormClosing(object sender, System.ComponentModel.Cance
...
+
+        private void RemoveButton_Click(object sender, EventArgs e)
+        {
+            try
+            {
+                if (itemListView.CheckedItems.Count > 0)
+                {
+                    using (TransactionScope scope = new TransactionScope())
+                    {
+                        List<Item> items = new List<Item>();
```
Comment (KarlZablan) [inline Forms/AddItemsForm.cs]: lines 201 - 206, please move it outside of the using statement.

File: Forms/CartForm.cs [hunk 6]
```diff
@@ -246,6 +246,19 @@ private void CheckoutButton_Click(object sender, EventArgs e)
                         _purchase.Date = DateTime.Now.ToString();
                         _purchase.Total = ComputeTotalPrice();
 
+                        foreach (PurchaseDetails purchaseDetails in _purchases)
```
Comment (KarlZablan) [inline Forms/CartForm.cs]: Optimize

File: Forms/ProfileForm.cs [hunk 7]
```diff
@@ -62,7 +62,7 @@ public void LoadData()
                     {
                         foreach (PurchaseDetails purchaseDetails in purchaseHistory.PurchaseDetails)
```
Comment (KarlZablan) [inline Forms/ProfileForm.cs]: line 63 - 68: Please optimize

**Review by KarlZablan — CHANGES_REQUESTED** [review]
Please check comments

File: Forms/CartForm.cs (hunk 6 above)
Comment (mllrdev-carlo-medel) [reply Forms/CartForm.cs]: Will remove this. This part was already handled in customer factory.

**Pull request merged**